packages/flutter/test/material/refresh_indicator_test.dart

```diff
@@ -1203,4 +1203,102 @@ void main() {
       await testColor(color);
     }
   });
+  testWidgets('RefreshIndicator default elevation value correctly',
+    (WidgetTester tester) async {
+  const List<double> positions = <double>[50.0, 100.0, 150.0];
+
+    final AnimationController positionController =
+        AnimationController(vsync: const TestVSync());
+    addTearDown(positionController.dispose);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: RefreshIndicator(
+          onRefresh: refresh,
+          child: ListView(
+            physics: const AlwaysScrollableScrollPhysics(),
+            children: const <Widget>[Text('X')],
+          ),
+        ),
+      ),
+    );
+
+    RefreshProgressIndicator getIndicator() {
+      return tester.widget<RefreshProgressIndicator>(
+        find.byType(RefreshProgressIndicator),
+      );
+    }
+
+    // Correspond to [_kDragContainerExtentPercentage].
+    final double maxPosition =
+        tester.view.physicalSize.height / tester.view.devicePixelRatio * 0.25;
+    for (final double position in positions) {
+      await tester.fling(find.text('X'), Offset(0.0, position), 1.0);
+      await tester.pump();
+      positionController.value = position / maxPosition;
+      expect(
+        getIndicator().elevation,
+        2.0,
+      );
+      // Wait until the fling finishes before starting the next fling.
+      await tester.pumpAndSettle();
+    }
+  });
+
+  testWidgets('RefreshIndicator custom elevation values correctly',
+      (WidgetTester tester) async {
+    const List<double> elevations = <double>[
+      0,
+      0.1,
+      1,
+      1.1,
+      1.59,
+    ];
+
+    const List<double> positions = <double>[50.0, 100.0, 150.0];
+
+    Future<void> testElevation(double elevation) async {
+      final AnimationController positionController =
+          AnimationController(vsync: const TestVSync());
+      addTearDown(positionController.dispose);
+
+      await tester.pumpWidget(
+        MaterialApp(
+          home: RefreshIndicator(
+            elevation: elevation,
+            onRefresh: refresh,
+            child: ListView(
+              physics: const AlwaysScrollableScrollPhysics(),
+              children: const <Widget>[Text('X')],
+            ),
+          ),
+        ),
+      );
+
+      RefreshProgressIndicator getIndicator() {
```

Comment:
As before, this function might be unneeded, serving to complicate instead of make easier.

Reply:
@victorsanni - Please check the above few test-cases which are already part of the master branch. On the other hand, I understand these are subjective and varies person to person. I feel perfectly fine reading the function because the naming is clear as the name suggest what it will resolve into. If there is a clear guide on writing a test case please share the link.

Reply:
> I understand these are subjective and varies person to person.
I do agree that these are subjective. I would look at it from the perspective of someone looking at the code for the first time. They would do something like this:

1. Get to `tester.fling`. The understanding of the viewer here is that a refresh is triggered.
2. Get to the `expect` block. Now, they would see the `getIndicator` function. While they would (most likely) guess that the tester is finding the `RefreshProgressIndicator` created in the test, there are also other possible guesses. Maybe the `getIndicator` function is calling some global indicator created somewhere else, possibly outside the scope of this specific test. Maybe its getting another widget class underneath RefreshProgressIndicator (a render object, for instance) instead of an instance of a `RefreshProgressIndicator`.

This ambiguity leads to more time wasted trying to figure out what's happening. We should strive to eliminate ambiguity whenever possible. 

> I feel perfectly fine reading the function because the naming is clear as the name suggest what it will resolve into.

We shouldn't think that because something seems clear to me, or you, it would also seem clear to others. Usually, that's not the case. In fact, something that seems clear to me right now might not seem clear to me after some time.

> Please check the above few test-cases which are already part of the master branch.
We shouldn't keep doing something that might not be right because that's how it's always been done. That leads to mistakes and bad practices multiplying and persisting. Instead, we should strive to do what's best in any given scenario.

I understand this specific use case seems trivial, but small positive changes add up over time to create a product/system that everyone that comes across loves. :)

Reply:
> 2. Get to the `expect` block. Now, they would see the `getIndicator` function. While they would (most likely) guess that the tester is finding the `RefreshProgressIndicator` created in the test, there are also other possible guesses. Maybe the `getIndicator` function is calling some global indicator created somewhere else, possibly outside the scope of this specific test. Maybe its getting another widget class underneath RefreshProgressIndicator (a render object, for instance) instead of an instance of a `RefreshProgressIndicator`.
@victorsanni - Honestly, I feel like you are stretching this a lot. This "getIndicator" function is just few lines above the code. If a person who is looking at the code findings it difficult to locate. I am afraid no one can help.

> We shouldn't keep doing something that might not be right because that's how it's always been done. That leads to mistakes and bad practices multiplying and persisting. Instead, we should strive to do what's best in any given scenario.
@victorsanni Are you suggesting, other reviewers were not good? That they were not able to point of the obvious issues. Which is an issue according to you! 

I already told you this approach is fine by me. Unless and until, I am not going against the code structure guidelines. 

Peace✌🏼

Reply:
@dishankjindal1 when and where to encapsulate code into functions is something every engineer has a personal style/preference for.

As you said, there isn't any formal rule in the style guide as to this, and so in no way am I trying to block your PR from being landed. I only hope that the reasons I listed helped clarify the pros and cons of having that function in this situation.

> Are you suggesting, other reviewers were not good? That they were not able to point of the obvious issues. Which is an issue according to you!

It is my responsibility as a reviewer to raise concerns I might have with regards to changes made in a PR. This is independent of the opinions of other reviewers. If I see something that I think doesn't look good, or could be better, I will articulate why I hold that view. As I have already said, I only hope the reasons I gave made sense to you, and to anyone else following the conversation. 

In the end, as you've mentioned, it is a trivial issue, and I want to reiterate that I'm not intentionally trying to block your PR from being landed. I'm only here to help, and I hope we can both learn and grow from this experience to be better engineers. Thank you!

Reply:
@victorsanni - I made the changes as requested. Everything looks good to me now.

Reply:
It does feel a little long. If you prefer, you can define the finder a line above to make it a little more readable. But all this is personal preference too. I should have LGTMed even before this conversation, my mistake.